qualtran/bloqs/basic_gates/global_phase.py

```diff
@@ -58,6 +77,27 @@ def my_tensors(
 
         return [qtn.Tensor(data=self.coefficient, inds=[], tags=[str(self)])]
 
+    def get_ctrl_system(
+        self, ctrl_spec: Optional['CtrlSpec'] = None
+    ) -> Tuple['Bloq', 'AddControlledT']:
+
+        # Delegate to superclass logic for more than one control.
+        if not (ctrl_spec is None or ctrl_spec == CtrlSpec()):
+            return super().get_ctrl_system(ctrl_spec=ctrl_spec)
+
+        # Otherwise, it's a ZPowGate
+        exponent: float = np.real_if_close(np.log(self.coefficient) / (np.pi * 1.0j)).item()  # type: ignore
```

Comment:
this wouldn't work when `self.coefficient` is symbolic. Maybe we should update this to `self.exponent` to match the other rotation gates?

Reply:
Thanks. 

Did you want to change the semantics to just store the exponent directly as the attribute or just change the name?

Reply:
the former, then we can safely pass `self.exponent` to `ZPowGate`. But would require updating all call sites. Based on a preliminary scan, most of them are `1j, -1j, -1` and would simplify SU2: https://github.com/quantumlib/Qualtran/blob/3315602fc762b3b94300d449aebe2a23fbddb29a/qualtran/bloqs/basic_gates/su2_rotation.py#L131

I'm happy to take this PR if you'd like.

Reply:
please :)